code/modules/mining/minebot.dm

```diff
@@ -101,25 +101,25 @@
 		to_chat(user, "<span class='info'>[src] is at full integrity.</span>")
 		return
 
-	if(I.use_tool(src, user, 0, volume=40))
+	if(welder.use_tool(src, user, 0, volume=40))
 		adjustBruteLoss(-15)
 		to_chat(user, "<span class='info'>You repair some of the armor on [src].</span>")
 
-/mob/living/simple_animal/hostile/mining_drone/attackby(obj/item/I, mob/user, params)
-	if(istype(I, /obj/item/mining_scanner) || istype(I, /obj/item/t_scanner/adv_mining_scanner))
+/mob/living/simple_animal/hostile/mining_drone/attackby(obj/item/item_used, mob/user, params)
+	if(istype(item_used, /obj/item/mining_scanner) || istype(item_used, /obj/item/t_scanner/adv_mining_scanner))
 		to_chat(user, "<span class='info'>You instruct [src] to drop any collected ore.</span>")
 		DropOre()
 		return
-	if(I.tool_behaviour == TOOL_CROWBAR || istype(I, /obj/item/borg/upgrade/modkit))
-		I.melee_attack_chain(user, stored_gun, params)
+	if(item_used.tool_behaviour == TOOL_CROWBAR || istype(item_used, /obj/item/borg/upgrade/modkit))
+		item_used.melee_attack_chain(user, stored_gun, params)
 		return
 	..()
 
 /mob/living/simple_animal/hostile/mining_drone/death()
-	DropOre(0)
+	DropOre()
 	if(stored_gun)
-		for(var/obj/item/borg/upgrade/modkit/M in stored_gun.modkits)
-			M.uninstall(stored_gun)
+		for(var/obj/item/borg/upgrade/modkit/modkit in stored_gun.modkits)
```

Comment:
```suggestion
		for(var/obj/item/borg/upgrade/modkit/modkit as anything in stored_gun.modkits)
```

unless there's for some reason something else in modkits

Reply:
What difference does `as anything` make?